### PR TITLE
Datasource settings: Add deprecation notice for database field

### PR DESCRIPTION
**Pull request by zoltanbedi**
This PR adds deprecation notice for the database field to let others know that we are migrating away from that field and instead should use the JSONData field for database related settings.

This is also based on our conversation with @marefr in slack. https://raintank-corp.slack.com/archives/C01C4K8DETW/p1668075734953779

Related to https://github.com/grafana/grafana-plugin-sdk-go/pull/555

### CODE REVIEW

File: pkg/services/datasources/service/store.go
Comment (marefr) [inline pkg/services/datasources/service/store.go:289]: Skeptic you want to do this before merging the other PR. Doesn't that risk clearing database column without writing database value to json data?

File: packages/grafana-data/src/types/datasource.ts
Comment (leventebalogh) [inline packages/grafana-data/src/types/datasource.ts:563]: Sorry if it was asked somewhere else already, but are we planning to remove the use of this from Elasticsearch and Influxdb as well?

Comment (zoltanbedi) [reply packages/grafana-data/src/types/datasource.ts:563]: Sorry but I don't know anything about them. Maybe @grafana/observability-logs can answer?

File: pkg/services/datasources/service/store.go
Comment (zoltanbedi) [reply pkg/services/datasources/service/store.go:289]: I can't really see this as a risk but we can also do this in the other pr. Just wanted to separate them a little.

File: packages/grafana-data/src/types/datasource.ts
Comment (gabor) [reply packages/grafana-data/src/types/datasource.ts:563]: we'd like to remove the use of this from Elasticsearch, but there's nothing planned yet (i created a github-issue to track this at https://github.com/grafana/grafana/issues/59098)

Comment (gabor) [reply packages/grafana-data/src/types/datasource.ts:563]: regarding Influxdb, pinging @grafana/observability-metrics 😄

Comment (gtk-grafana) [reply packages/grafana-data/src/types/datasource.ts:563]: I don't know of any plan to deprecate this, but it doesn't look like a huge lift to follow the plan laid out in the issue Gabor created above. I created an issue here https://github.com/grafana/grafana/issues/59115, so we don't loose track of this request.

File: pkg/services/datasources/service/store.go
Comment (marefr) [reply pkg/services/datasources/service/store.go]: Merging this PR would trigger a nightly build (at least used to do) and/or grafana cloud build so there's a chance this creates an actual problem if we're unlucky. Would suggest doing it in the other PR or merge them together

Comment (marefr) [reply pkg/services/datasources/service/store.go]: Sorry for slow response btw. Just realized you've removed this now so will approve

**Review by marefr — COMMENTED** [review]
LGTM, but think @grafana/plugins-platform-frontend should approve this

**Review by leventebalogh — APPROVED** [review]
🚢🇮🇹

**Pull request merged**
